### PR TITLE
lsp names renamed to roc_language_server

### LINKED ISSUE #6330: Rename all instances of `roc_ls` and `roc_lang_server` to `roc_language_server`
Descriptive names are better :)

### PR DESCRIPTION
**Pull request by B-Teague**
Renamed instances of roc_ls and roc_lang to roc_language_server.

Closes #6330

Blocked by #6354

### CODE REVIEW

File: crates/cli_testing_examples/benchmarks/CFold
Comment (lukewilliamboswell) [inline crates/cli_testing_examples/benchmarks/CFold:1]: Should these files be here I wonder?

Comment (B-Teague) [reply crates/cli_testing_examples/benchmarks/CFold:1]: They probably got updated when I ran ```cargo test --release```
Do we need to add these to .gitignore?

Comment (B-Teague) [reply crates/cli_testing_examples/benchmarks/CFold:1]: The gitignore case did not match the executable file names.

Probably should globalize .gitignore more so we can ignore all binary files at the root of the project.



# Ignore everything including files without extensions (binary files)
*
# Unignore all files with extensions
!*.*
# Unignore all directories
!*/

# Add specific ignore files here

**Comment by Anton-4** [comment]
Thanks for your contribution @B-Teague :)
Could you sign this commit as well and resolve the merge conflict?

**Comment by Anton-4** [comment]
Can you merge this @rtfeldman?

**Pull request merged**